modules/tinymce/src/themes/silver/main/ts/ui/dialog/Collection.ts

```diff
@@ -113,6 +118,18 @@ export const renderCollection = (spec: CollectionSpec, providersBackstage: UiFac
     components: [ ],
     factory: { sketch: Fun.identity },
     behaviours: Behaviour.derive([
+      Disabling.config({
+        disabled: providersBackstage.isReadonly(),
+        onDisabled: (comp) => {
+          Arr.map(SelectorFilter.descendants(comp.element(), '.tox-collection__item'), (childElm) => Class.add(childElm, 'tox-collection__item--state-disabled'));
```

Comment:
I mentioned this IRL, but looks like it was forgotten but there's a lot of duplicate code here that should be pulled out. It's also very inefficient as you're looping over the same nodes twice which isn't needed.

- [ ] Pull iteration function out to remove duplicate code
- [ ] Don't iterate over the child nodes twice to apply the style/attributes

Reply:
As an example, this could be something like this:
```
onDisabled: (comp) => {
  iterChildItems((childElm) => {
    Class.add(childElm, 'tox-collection__item--state-disabled'));
    Attr.set(childElm, 'aria-disabled', true);
  });
}
```

Reply:
Yes, sorry I forgot to do it. Thanks for pointing it out.